controller/controlplane/controller_watch.go

```diff
@@ -0,0 +1,237 @@
+package controlplane
+
+import (
+	"context"
+	"reflect"
+
+	admregv1 "k8s.io/api/admissionregistration/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	operatorerrors "github.com/kong/gateway-operator/internal/errors"
+	"github.com/kong/gateway-operator/internal/utils/index"
+	"github.com/kong/gateway-operator/pkg/consts"
+
+	operatorv1beta1 "github.com/kong/kubernetes-configuration/api/gateway-operator/v1beta1"
+)
+
+func (r *Reconciler) clusterRoleHasControlPlaneOwner(obj client.Object) bool {
+	ctx := context.Background()
+
+	clusterRole, ok := obj.(*rbacv1.ClusterRole)
+	if !ok {
+		ctrllog.FromContext(ctx).Error(
+			operatorerrors.ErrUnexpectedObject,
+			"failed to run predicate function",
+			"expected", "ClusterRole", "found", reflect.TypeOf(obj),
+		)
+		return false
+	}
+
+	return r.clusterScopedObjHasControlPlaneOwner(ctx, clusterRole)
+}
+
+func (r *Reconciler) clusterRoleBindingHasControlPlaneOwner(obj client.Object) bool {
+	ctx := context.Background()
+
+	clusterRoleBinding, ok := obj.(*rbacv1.ClusterRoleBinding)
+	if !ok {
+		ctrllog.FromContext(ctx).Error(
+			operatorerrors.ErrUnexpectedObject,
+			"failed to run predicate function",
+			"expected", "ClusterRoleBinding", "found", reflect.TypeOf(obj),
+		)
+		return false
+	}
+
+	return r.clusterScopedObjHasControlPlaneOwner(ctx, clusterRoleBinding)
+}
```

Comment:
Nit: these 2 + `validatingWebhookConfigurationHasControlPlaneOwner` could be made to use generics to prevent code duplication.